src/commonTest/kotlin/app/cash/turbine/FlowInScopeTest.kt

```diff
@@ -208,7 +208,9 @@ class FlowInScopeTest {
         delay(1100.milliseconds)
       }
     }.testIn(this, timeout = 1500.milliseconds)
-    turbine.awaitComplete()
+    withContext(Default) {
```

Comment:
Adding these `withContext(Default)` calls here defeats the entire purpose of these tests. 

In any of these tests that uses `withContext(Default)` within the test subject, the test subject is simulating the experience of an engineer whose code is broken.

If we also run `awaitComplete()` on the `Default` dispatcher, we're no longer simulating the experience of that engineer, becuase they won't be calling `awaitComplete()` on `Default`: they'll be calling it on `TestScope`'s dispatcher.

This is a critical service that Turbine provides: unlike `runTest`'s timeout, Turbine provides specific feedback on which expected output failed to emit.

Reply:
> In any of these tests that uses withContext(Default) within the test subject, the test subject is simulating the experience of an engineer whose code is broken.

To clarify this: it simulates this because the engineer literally has a test that is hanging awaiting some value, and their wall clock is ticking. 

To me, an interesting test for your change would be something like this:

```kotlin
@Test fun awaitFailsOnVirtualTime() = runTestTurbine {
  assertFailsWith<AssertionError> {
    flow<Nothing> {
      delay(1500.milliseconds)
    }.test(timeout = 1000.milliseconds) {
      awaitComplete()
    }
  }
}
```

Or:

```kotlin
@Test fun awaitFailsOnVirtualTime() = runTestTurbine {
  assertFailsWith<AssertionError> {
    flow<Nothing> {
      awaitCancellation()
    }.test {
      awaitComplete()
    }
  }
}
```

...in other words, can we just ditch the wallclock time stuff and rely entirely on virtual time?

Reply:
Thanks! I really had no idea what the broken tests were for.

7ba8b1fa078adbd78e8bcc65a2cdaefefec6e57e adds both tests you suggested, both passing.

Reply:
> can we just ditch the wallclock time stuff and rely entirely on virtual time?

That would be really nice 😄  and also what I'm hoping for.